File: sdk/php/examples/translate.php
<?php
/**
 * Source Translator - PHP Example
 * 
 * Usage:
 *   php examples/translate.php "Hello, World!" en pt-BR
 *   php examples/translate.php "Olá mundo" pt-AO en
 */

require_once __DIR__ . '/../vendor/autoload.php';

use SourceTranslator\SourceTranslator;

// Get arguments from command line
$text = $argv[1] ?? 'Hello, World!';
$sourceLang = $argv[2] ?? 'en';
$targetLang = $argv[3] ?? 'pt-BR';

echo "Initializing Source Translator...\n";
$translator = new SourceTranslator();

echo "Translating: \"{$text}\"\n";
echo "From: {$sourceLang} -> To: {$targetLang}\n\n";

try {
    $result = $translator->translate($text, $sourceLang, $targetLang);

    echo "Result:\n";
    echo "  Translation: \"{$result['translated_text']}\"\n";
    echo "  Provider: {$result['provider']}\n";
    echo "  Cached: " . ($result['cached'] ? 'true' : 'false') . "\n";
    echo "  Latency: {$result['latency_ms']}ms\n\n";

    echo "Cache Stats:\n";
    $stats = $translator->cacheStats();
    echo "  Total entries: {$stats['total_entries']}\n";
    echo "  Providers used: " . implode(', ', $stats['providers_used']) . "\n";
} catch (\Exception $e) {
    echo "Error: " . $e->getMessage() . "\n";
    exit(1);
}

File: core/examples/translate.rs
use source_translator_core::{Translator, TranslateRequest};

#[tokio::main]
async fn main() {
    println!("Initializing Source Translator...");
    
    let translator = match Translator::new().await {
        Ok(t) => t,
        Err(e) => {
            eprintln!("Failed to initialize: {}", e);
            return;
        }
    };
    
    println!("Translator ready!\n");
    
    // Test single translation
    let request = TranslateRequest {
        text: "Hello, World!".to_string(),
        source_lang: "en".to_string(),
        target_lang: "pt-BR".to_string(),
    };
    
    println!("Translating: \"{}\"", request.text);
    println!("From: {} -> To: {}", request.source_lang, request.target_lang);
    
    match translator.translate(request).await {
        Ok(result) => {
            println!("\nResult:");
            println!("  Translation: \"{}\"", result.translated_text);
            println!("  Provider: {}", result.provider);
            println!("  Cached: {}", result.cached);
            println!("  Latency: {}ms", result.latency_ms);
        }
        Err(e) => {
            eprintln!("Translation failed: {}", e);
        }
    }
    
    // Test cache stats
    println!("\nCache Stats:");
    match translator.cache_stats().await {
        Ok(stats) => {
            println!("  Total entries: {}", stats.total_entries);
            println!("  Providers used: {:?}", stats.providers_used);
        }
        Err(e) => {
            eprintln!("Failed to get stats: {}", e);
        }
    }
}
